src/constraint_system/arithmetic.rs

```diff
@@ -74,7 +74,9 @@ impl StandardComposer {
         self.q_fixed_group_add.push(BlsScalar::zero());
         self.q_variable_group_add.push(BlsScalar::zero());
 
-        self.public_inputs.push(pi);
+        if let Some(pi) = pi {
+            assert!(self.public_inputs_sparse_store.insert(self.n, pi).is_none());
```

Comment:
I believe we should go in a direction to reduce the number of assertions inside plonk functions, not increase it. If the PI must fail in this case, then the function should return a `Result`

Reply:
This should never happen. That's why it has an `assert!` it's like `unreachable!()` for me here. I just have it so that if at some point this assertion fails. We know that we made a change in a really wrong direction.